gdal/ogr/ogrsf_frmts/gpkg/ogrgeopackagedatasource.cpp

```diff
@@ -5156,6 +5156,24 @@ OGRErr GDALGeoPackageDataset::DeleteRasterLayer( const char* pszLayerName )
         sqlite3_free(pszSQL);
     }
 
+    if( eErr == OGRERR_NONE )
+    {
+        char* pszSQL = sqlite3_mprintf(
+                "DELETE FROM gpkg_2d_gridded_coverage_ancillary WHERE lower(tile_matrix_set_name) = lower('%q')",
```

Comment:
the existence of this table should be verified, otherwise this is going to error out for regular "visual" rasters. To do so, you can extract the code at the beginning of GDALGeoPackageDataset::CreateTileGriddedTable() into a dedicated ::HasGriddedCoverageAncillaryTable() method that can be used there and here.